typing/typecore.ml

```diff
@@ -4050,21 +4045,10 @@ let type_omitted_parameters expected_mode env loc ty_ret mode_ret args =
                newty2 ~level
                  (Tarrow (arrow_desc, ty_arg, ty_ret, commu_ok))
              in
-             let new_closed_args =
-               List.map
-                 (fun (exp, marg) ->
-                    submode ~loc:exp.exp_loc ~env ~reason:Other
-                      marg (mode_partial_application expected_mode);
-                    value_to_alloc_r2l marg)
-                 open_args
-             in
-             let closed_args = new_closed_args @ closed_args in
-             let open_args = [] in
-             let mode_closed_args = List.map Alloc.close_over closed_args in
              let mode_partial_fun = Alloc.partial_apply mode_fun in
              let mode_closure, _ =
                Alloc.newvar_above (Alloc.join
-                (mode_partial_fun:: mode_closed_args))
+                [ mode_partial_fun; closed_args_mode ])
```

Comment:
I suspect it's clearer to use `Value` (instead of `Alloc`) throughout this function, since many of those modes are not allocation and thus not suitable for `Alloc`. Then, the following call to `register_allocation_mode` can be replaced by `register_allocation`.

Reply:
After second thought, I think the current code is fine - since we are in the same region anyway (since those psudo-functions don't have region).